urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/INDIDate.java

```diff
@@ -0,0 +1,261 @@
+   * 
+   * @param years The years component of the Date. All values are valid.
+   * @param months The months component of the Date. Mut be in the range 1
+   * (January) - 12 (December).
+   * @param days The days component of the Date. Must be in the 1-28,29,30,31
+   * range (depending on the months and years).
+   */
+  public INDIDate(int years, int months, int days) {
+    initializeComponents();
+    
+    setYears(years);
+    setMonths(months);
+    setDays(days);
+  }
+  
+  /**
+   * Constructs a new Date whose components are partially specified. Hours, 
+   * minutes and seconds are initialized at 00:00:00.0.
+   * 
+   * @param years The years component of the Date. All values are valid.
+   * @param months The months component of the Date. Mut be in the range 1
+   * (January) - 12 (December).
+   * @param days The days component of the Date. Must be in the 1-28,29,30,31
+   * range (depending on the months and years).
+   * @param hours The hours component of the Date. Must be in the 0-23 range.
+   * @param minutes The minutes component of the Date. Must be in the 0-59
+   * range.
+   * @param seconds The seconds component of the Date. Must be in the
+   * 0-59.999999 range.
+   */
+  public INDIDate(int years, int months, int days, int hours, int minutes, double seconds) {
+    initializeComponents();
+    
+    setYears(years);
+    setMonths(months);
+    setDays(days);
+    setHours(hours);
+    setMinutes(minutes);
+    setSeconds(seconds);
+  }
+  
+  /**
+   * Initilizes the components to 0000/01/01 00:00:00.0
+   */
+  private void initializeComponents() {
+    years = 0;
+    months = 1;
+    days = 1;
+    hours = 0;
+    minutes = 0;
+    seconds = 0;
+  }
+  
+  /**
+   * Gets the years component of the Date.
+   *
+   * @return The years component of the Date.
+   */
+  public int getYears() {
+    return years;
+  }
+
+  /**
+   * Sets the years component of the Date.
+   *
+   * @param years The years component of the Date. All values are valid.
+   */
+  public final void setYears(int years) {
+    this.years = years;
+  }
+
+  /**
+   * Gets the months component of the Date.
+   *
+   * @return The months component of the Date.
+   */
+  public int getMonths() {
+    return months;
+  }
+
+  /**
+   * Sets the months component of the Date.
+   *
+   * @param months The months component of the Date. Mut be in the range 1
+   * (January) - 12 (December).
+   */
+  public final void setMonths(int months) {
+    if ((months >= 0) && (months < 60)) {
+      this.months = months;
+    }
+  }
+
+  /**
+   * Gets the days component of the Date.
+   *
+   * @return The days component of the Date.
+   */
+  public int getDays() {
+    return days;
+  }
+
+  /**
+   * Sets the days component of the Date.
+   *
+   * @param days The days component of the Date. Must be in the 1-28,29,30,31
+   * range (depending on the months and years).
+   */
+  public final void setDays(int days) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.